06_deploy_bootstrap_vm.sh

```diff
@@ -6,63 +6,25 @@ source ocp_install_env.sh
 source common.sh
 source utils.sh
 
-# FIXME this is configuring for the libvirt backend which is dev-only ref
-# https://github.com/openshift/installer/blob/master/docs/dev/libvirt-howto.md
-# We may need some additional steps from that doc in 02* and also to make the
-# qemu endpoint configurable?
 if [ ! -d ocp ]; then
     mkdir -p ocp
-    cat > ocp/install-config.yaml << EOF
+    cat > ocp/baremtal-install-config.yaml << EOF
 apiVersion: v1beta3
 baseDomain: ${BASE_DOMAIN}
 metadata:
   name: ${CLUSTER_NAME}
 platform:
-  libvirt: {}
+  baremetal: {}
 pullSecret: |
   ${PULL_SECRET}
 sshKey: |
   ${SSH_PUB_KEY}
 EOF
 fi
 
-
-$GOPATH/src/github.com/openshift/installer/bin/openshift-install --dir ocp --log-level=debug create ignition-configs
-
-# Here we can make any necessary changes to the ignition configs/manifests
-# they can later be sync'd back into the installer via a new baremetal target
-
-# Now re create the cluster (using the existing install-config and ignition-configs)
-# Since we set the replicas to null, we should only get the bootstrap VM
-# FIXME(shardy) this doesn't work, it creates the bootstrap and one master
-#$GOPATH/src/github.com/openshift/installer/bin/openshift-install --dir ocp --log-level=debug create cluster
-#exit 1
-
-# ... so for now lets create the bootstrap VM manually and use the generated ignition config
-# See https://coreos.com/os/docs/latest/booting-with-libvirt.html
-IGN_FILE="/var/lib/libvirt/images/${CLUSTER_NAME}-bootstrap.ign"
-sudo cp ocp/bootstrap.ign ${IGN_FILE}
-
-# Apply patches to bootstrap ignition
-apply_ignition_patches bootstrap "$IGN_FILE"
```

Comment:
I was chatting with @celebdor about this, we'll need a way to merge in the metalkube additions to the ignition config, either by adding specific assets to the fork of openshift-install, or maybe as an interim step we allow for merging arbitrary json data like we're doing already in dev-scripts?

Reply:
Yeah, I added the keepalived ignition config change to kni-install as an example: https://github.com/metalkube/kni-installer/commit/1a7e01e15a6775f09866c3eee02a72233821218a

Reply:
Ah nice, that fits pretty cleanly, I was originally imagining something where we conditionally generated the assets for a given platform (which I guess we may still need to do..)

Reply:
Yes - I had a FIXME in the commit message about that. Do we need to make it conditional on the platform? Do other platforms ever do that?

Reply:
OpenStack does a bit of ignition customization, but it's just via terraform.

https://github.com/openshift/installer/blob/master/data/data/openstack/bootstrap/main.tf#L14

All of the core ignition config is passed in as a variable.  That data goes into Swift.  Later the ignition config given to the bootstrap VM is defined as both the URL to the core bootstrap config in Swift, and a custom ifcfg file.

Reply:
So actually, if you told the installer to generate the ignition configs for OpenStack, you wouldn't see those custom changes ...

Reply:
@celebdor note this would probably replace the env merging stuff, if we can just add the files needed and/or template them.

Reply:
I was actually thinking about the go generated ignition stuff e.g https://github.com/openshift/installer/blob/master/pkg/asset/ignition/bootstrap/bootstrap.go where we could probably have some conditional parts added per platform.

I like the idea of the static files though, maybe we could just add logic to consume per-platform directories so we don't risk changing stuff for existing platforms?

Reply:
Captured the point about bare metal specific ignition modifications here - https://github.com/metalkube/kni-installer/blob/master/docs/dev/baremetal.md